### PR TITLE
api,config: enable audit-middleware in default config

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

default enable audit

Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: ref #4480

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
image

- Benchmark test result
BenchmarkDoRequestWithLocalLogAudit
> 5206        211411 ns/op       26330 B/op        190 allocs/op
	   6152        247705 ns/op       26319 B/op        190 allocs/op

BenchmarkDoRequestWithPrometheusAudit
> 4790        351855 ns/op       31247 B/op        245 allocs/op
	  4512        276509 ns/op       30732 B/op        240 allocs/op

BenchmarkDoRequestWithoutServiceMiddleware
> 5767        237742 ns/op       26352 B/op        190 allocs/op
	  4887        209907 ns/op       26329 B/op        190 allocs/op

Code changes

- Has configuration change

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: metrics/grafana/pd.json [hunk 1]
```diff
@@ -8326,6 +8326,216 @@
...
+              "legendFormat": "{{service}}-{{ip}}",
+              "refId": "A",
+              "step": 4
+            }
+          ],
+          "thresholds": [],
+          "timeFrom": null,
+          "timeRegions": [],
+          "timeShift": null,
+          "title": "Completed commands rate",
```
Comment (rleungx) [inline metrics/grafana/pd.json:8389]: How about adding HTTP prefix？

File: server/api/router.go [hunk 2]
```diff
@@ -162,20 +162,20 @@ func createRouter(prefix string, svr *server.Server) *mux.Router {
...
+	registerFunc(apiRouter, "/config/cluster-version", confHandler.SetClusterVersion, setMethods(http.MethodPost), setAuditBackend(localLog))
 	registerFunc(apiRouter, "/config/replication-mode", confHandler.GetReplicationModeConfig, setMethods(http.MethodGet))
-	registerFunc(apiRouter, "/config/replication-mode", confHandler.SetReplicationModeConfig, setMethods(http.MethodPost))
+	registerFunc(apiRouter, "/config/replication-mode", confHandler.SetReplicationModeConfig, setMethods(http.MethodPost), setAuditBackend(localLog))
 
 	rulesHandler := newRulesHandler(svr, rd)
 	registerFunc(clusterRouter, "/config/rules", rulesHandler.GetAllRules, setMethods(http.MethodGet))
-	registerFunc(clusterRouter, "/config/rules", rulesHandler.SetAllRules, setMethods(http.MethodPost), setAuditBackend(localLog))
-	registerFunc(clusterRouter, "/config/rules/batch", rulesHandler.BatchRules, setMethods(http.MethodPost), setAuditBackend(localLog))
+	registerFunc(clusterRouter, "/config/rules", rulesHandler.SetAllRules, setMethods(http.MethodPost), setAuditBackend(localLog), setAuditBackend(prometheus))
```
Comment (rleungx) [inline server/api/router.go:173]: Why change it?

File: server/api/router.go [hunk 3]
```diff
@@ -243,48 +243,48 @@ func createRouter(prefix string, svr *server.Server) *mux.Router {
...
 	registerFunc(apiRouter, "/version", newVersionHandler(rd).GetVersion, setMethods(http.MethodGet))
 	registerFunc(apiRouter, "/status", newStatusHandler(svr, rd).GetPDStatus, setMethods(http.MethodGet))
 
 	memberHandler := newMemberHandler(svr, rd)
-	registerFunc(apiRouter, "/members", memberHandler.GetMembers, setMethods(http.MethodGet))
-	registerFunc(apiRouter, "/members/name/{name}", memberHandler.DeleteMemberByName, setMethods(http.MethodDelete), setAuditBackend(localLog))
-	registerFunc(apiRouter, "/members/id/{id}", memberHandler.DeleteMemberByID, setMethods(http.MethodDelete), setAuditBackend(localLog))
-	registerFunc(apiRouter, "/members/name/{name}", memberHandler.SetMemberPropertyByName, setMethods(http.MethodPost), setAuditBackend(localLog))
+	registerFunc(apiRouter, "/members", memberHandler.GetMembers, setMethods(http.MethodGet), setAuditBackend(prometheus))
+	registerFunc(apiRouter, "/members/name/{name}", memberHandler.DeleteMemberByName, setMethods(http.MethodDelete), setAuditBackend(localLog), setAuditBackend(prometheus))
```
Comment (rleungx) [inline server/api/router.go:277]: ditto

**Comment by rleungx** [comment]
BTW, I suggest adding a comment to explain the principle of using logs or metrics.

File: server/api/router.go (hunk 2 above)
Comment (CabinfeverB) [reply server/api/router.go:174]: add metrics for all API

Comment (rleungx) [reply server/api/router.go:174]:
```suggestion
	registerFunc(clusterRouter, "/config/rules", rulesHandler.SetAllRules, setMethods(http.MethodPost), setAuditBackend(localLog, prometheus))
```

**Review by rleungx — APPROVED** [review]

**Pull request merged**